java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java

```diff
@@ -202,22 +221,48 @@ public static URI urlToFSPath(String urlString, String name) {
             host = url.getHost();
             path = url.getPath();
 
+            // Case 1
             if ("localhost".equals(host)) {
                 return url.toURI();
             }
+            String qPath = Pattern.compile("&")
+                    .splitAsStream(Optional.ofNullable(url.getQuery()).orElse(""))
+                    .filter(f -> f.contains("=")) // filter out possible auth tokens
+                    .map(s -> Arrays.copyOf(s.split("=", 2), 2))
```

Comment:
`copyOf` should not be needed here since the filter makes sure there is always a `=` and split will always return 2 elements with empty string in case there is nothing after/before `=`

Reply:
i think this should be equivalent.
```java
      Arrays.stream(url.getQuery().split("&"))
                .filter(p -> p.contains("="))
                .map(p ->
                    Arrays.stream(p.split("=", 2))
                        .map(s -> URLDecoder.decode(s, StandardCharsets.UTF_8))
                        .collect(joining("/"))
                )
                .sorted()
                .collect(Collectors.joining("/"));
 ```

Reply:
Trying out

Reply:
changed